test/unit/range/iterator_test_template.hpp

```diff
@@ -685,35 +685,43 @@ inline void address_difference_test(it_begin_t it_begin, it_sentinel_t it_end)
 
 TYPED_TEST_P(iterator_fixture, address_of_end)
 {
-    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    // sentinel must not be the same as the iterator
+    if constexpr (std::ranges::common_range<decltype(this->test_range)>)
     {
-        using difference_t = std::iter_difference_t<decltype(std::ranges::begin(this->test_range))>;
-        auto it_begin = std::ranges::begin(this->test_range);
-        auto it_end = std::ranges::end(this->test_range);
-        EXPECT_EQ(std::to_address(it_end), std::to_address(it_begin) + static_cast<difference_t>(it_end - it_begin));
-    }
+        if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+        {
+            using difference_t = std::iter_difference_t<decltype(std::ranges::begin(this->test_range))>;
+            auto it_begin = std::ranges::begin(this->test_range);
+            auto it_end = std::ranges::end(this->test_range);
+            EXPECT_EQ(std::to_address(it_end), std::to_address(it_begin) + static_cast<difference_t>(it_end - it_begin));
+        }
 
-    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
-                  TestFixture::const_iterable)
-    {
-        using difference_t = std::iter_difference_t<decltype(std::ranges::cbegin(this->test_range))>;
-        auto it_cbegin = std::ranges::cbegin(this->test_range);
-        auto it_cend = std::ranges::cend(this->test_range);
-        EXPECT_EQ(std::to_address(it_cend), std::to_address(it_cbegin) + static_cast<difference_t>(it_cend - it_cbegin));
+        if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
+                      TestFixture::const_iterable)
+        {
+            using difference_t = std::iter_difference_t<decltype(std::ranges::cbegin(this->test_range))>;
+            auto it_cbegin = std::ranges::cbegin(this->test_range);
+            auto it_cend = std::ranges::cend(this->test_range);
+            EXPECT_EQ(std::to_address(it_cend), std::to_address(it_cbegin) + static_cast<difference_t>(it_cend - it_cbegin));
+        }
     }
 }
 
 TYPED_TEST_P(iterator_fixture, address_difference)
 {
     if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
     {
-        address_difference_test(std::ranges::begin(this->test_range), std::ranges::end(this->test_range));
+        auto it = std::ranges::begin(this->test_range);
+        auto sentinel_it = std::ranges::next(it, std::ranges::end(this->test_range));
+        address_difference_test(it, sentinel_it);
```

Comment:
Shouldn't both the old and new version work?

Reply:
I have to investigate again :)

Reply:
Maybe I should add the reason: The only requirement for contiguous range is the following https://eel.is/c++draft/range.refinements#3

```
🔗
template<class T>
  concept contiguous_­range =
    random_­access_­range<T> && contiguous_­iterator<iterator_t<T>> &&
    requires(T& t) {
      { ranges::data(t) } -> same_­as<add_pointer_t<range_reference_t<T>>>;
    };
```

and

```
Given an expression t such that decltype((t)) is T&, T models contiguous_­range only if to_­address(​ranges​::​begin(t)) == ranges​::​data(t) is true.
```

that means it only required on the iterator (= std::ranges::begin) and not the sentinel.

Reply:
This expression here transforms any sentinel type into an iterator type.

Reply:
Makes sense :) 
Maybe add a small comment such as 

`The standard only requires the iterator, but not the sentinel, to be contiguous.`